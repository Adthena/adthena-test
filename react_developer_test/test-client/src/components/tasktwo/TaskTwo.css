button {
  right: 0;
  margin: 10px;
  position: absolute;
}

input,
.user-info,
.todo-list {
  width: 40%;
  height: 2;
  height: 30px;
  display: block;
  margin: 0 auto;
  text-align: center;
  margin-top: 50px;
}

.content {
  margin-top: 40px;
}
