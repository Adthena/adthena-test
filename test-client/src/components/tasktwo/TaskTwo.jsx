import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import dao from '@services/dao';

const TaskTwo = ({ message }) => {
  const [isLoading, setIsLoading] = useState(true);
  const [hasError, setHasError] = useState(false);
  const [data, setData] = useState(null);

  useEffect(() => {
    dao.GET(`/api/status?message=${message}`)
      .then((res) => {
        setData(res.message);
        setHasError(false);
        setIsLoading(false);
      })
      .catch(() => {
        setIsLoading(false);
        setHasError(true);
      });
  }, [message]);

  const title = `Hello ${data}!`;
  return (
    <div className="task">
      {hasError && (<h3>Server error...</h3>)}
      {isLoading && (<h3>Loading...</h3>)}
      {data && (
        <>
          <h1>{title}</h1>
          <div>
            {
              `As an example of how to make a request to the API, this message was generated 
              using data from the status endpoint.`
            }
          </div>
        </>
      )}
    </div>
  );
};

TaskTwo.propTypes = {
  message: PropTypes.string,
};

TaskTwo.defaultProps = {
  message: 'Adthena',
};

export default TaskTwo;
