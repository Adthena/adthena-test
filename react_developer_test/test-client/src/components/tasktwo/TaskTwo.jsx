import React, { useState, useEffect } from 'react';
import TodoList from './components/TodoList';
import UserSearch from './components/UserSearch';
import useFetch from './hooks/useFetch';
import DarkModeSwitch from './components/DarkModeSwitch'
import UserInfo from './components/UserInfo'
import { ThemeContext } from './components/ThemeContext';
import { createGlobalStyle } from 'styled-components';
import "./css/TaskTwo.css";


const TaskTwo = () => {
  const [userName, setUserName] = useState(null);
  const [userId, setUserId] = useState(null);
  const [user, errorSearchUser] = useFetch(`https://jsonplaceholder.typicode.com/users?username=${userName}`, "username");
  const [todos, errorSearchTodos] = useFetch(`https://jsonplaceholder.typicode.com/todos?userId=${userId}`, "userId");
  const [mode, setMode] = useState('light')


  const GlobalStyle = createGlobalStyle`
  body {
    background: ${mode === 'dark' ? 'black' : 'white'};
    color: ${mode === 'dark' ? 'white' : 'black'};
  }
`

  const handleSearch = (value) => {
    if (value) { setUserName(value); }
  }

  useEffect(() => {
    if (user && user.length) {
      setUserId(user[0].id);
    }
    else setUserId(null);
  }, [user]);



  return (
    <ThemeContext.Provider value={{ mode, 'toggleMode': setMode }} >
      <GlobalStyle />
      <DarkModeSwitch />
      <div className="task-two">
        <div className="content">
          <UserSearch handleSearch={handleSearch} error={errorSearchUser} />
          {user && < UserInfo user={user} />}
          {todos && <TodoList todos={todos} error={errorSearchTodos} />}
        </div>
      </div>
    </ThemeContext.Provider>
  );
}
export default TaskTwo;
