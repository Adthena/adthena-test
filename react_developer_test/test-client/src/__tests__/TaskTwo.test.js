import React from 'react';
import { render } from "enzyme";
import TaskTwo from '../components/tasktwo/TaskTwo';

it('renders correctly', () => {
    const rendered = render(<TaskTwo />)
    expect(rendered).toMatchSnapshot();
})
