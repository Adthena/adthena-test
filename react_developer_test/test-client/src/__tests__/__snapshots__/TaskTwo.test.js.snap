// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <button
    class="toggle-mode"
  >
    Toggle Dark Mode
  </button>,
  <div
    class="task-two"
  >
    <div
      class="content"
    >
      <div>
        <form>
          <input
            class="user-search"
            placeholder="search for a user tolist..."
            type="text"
            value=""
          />
        </form>
      </div>
      <div />
      <div
        class="todo-list"
      />
    </div>
  </div>,
]
`;
